### ISSUE TITLE
tools.download() (uploader_downloader) : FTP support ?

### ISSUE THREAD

**Issue by vhuber** [conan-io/conan#1326]
One of the archive I have to work with is on a FTP server.
Is there a planning to let Conan mange FTP connections ?

**Comment by memsharded** [comment]
I think it might be already be feasible to implement using the builtin ftp library in python: https://docs.python.org/2/library/ftplib.html

Probably (not tested), something like:
```python
import ftplib 
with ftplib.FTP("your_ip", "login", "password") as ftp:
    ftp.cwd('somefolder')
    with open("yourfilename.zip", 'wb') as f:
        ftp.retrbinary('RETR ' + "yourfilename.zip", f.write)
```

Have you tried something like this? Maybe a conan built-in helper might help, but could be doable today without having to wait.

**Comment by vhuber** [comment]
yes, I have tried to do this and - with no surprise - it works.

The question is actually about a built-in helper. Do you plan to implement it ?

**Comment by memsharded** [comment]
Yes, sure, makes sense and it is easy to do. Lets do it for next (0.24 release). Maybe you could contribute a PR yourself :) ?

**Comment by memsharded** [comment]
Released in 0.24
